backend/remote-state/s3/client.go

```diff
@@ -28,7 +35,58 @@ type RemoteClient struct {
 	lockTable            string
 }
 
-func (c *RemoteClient) Get() (*remote.Payload, error) {
+var (
+	// The amount of time we will retry a state waiting for it to match the
+	// expected checksum.
+	consistencyRetryTimeout = 10 * time.Second
+
+	// delay when polling the state
+	consistencyRetryPollInterval = 2 * time.Second
+
+	// checksum didn't match the remote state
+	errBadChecksum = errors.New("invalid state checksum")
```

Comment:
Do you think there's something better we could say here? Based on what you'd said to me before, it seemed like it's expected for S3 to occasionally take longer than 10 seconds to converge which would suggest this is an error message users would actually see in normal use, so ideally this message would include both an understandable statement of the problem _and_ a suggested next step.

Perhaps:

```go
errBadChecksumFmt = `state data in S3 does not have the expected content.

This may be caused by unusually-long delays in S3 processing a previous state update.
Please wait for a minute or two and try again. If this problem persists, and neither S3 nor
DynamoDB are experiencing an outage, the checksum stored in the DynamoDB table
may need to be manually updated to the following value:
    %x
`
```

(Where that %x marker would be passed `expected` as its value below.)

This approach has the disadvantage of not using a consistent error instance for all cases and thus making it harder for the caller to recognize the error, so if that's a concern then of course we could equally do a custom error type that has the expected state checksum as a field. That would then allow us to keep the error message itself simple and produce the more verbose error message at a higher layer, which may be preferable.